lib/jekyll-github-metadata.rb

```diff
@@ -61,9 +62,18 @@ def client
         @client ||= Client.new
       end
 
+      def repository
+        @repository ||= GitHubMetadata::Repository.new(nwo).tap do |repo|
+          Jekyll::GitHubMetadata.log :debug, "Generating for #{repo.nwo}"
+        end
+      end
+
+      def nwo
+        @nwo ||= GitHubMetadata::RepositoryFinder.nwo
```

Comment:
My struggle is that it's too easy for this data to get all out of sync. I want the behavior that if `GitHubMetadata.site` changes, that `GitHubMetadata.repository` changes with it.

```ruby
module GitHubMetadata
  def self.site
    repository_finder.site
  end

  def self.site=(new_site)
    reset!
    @repository_finder = GitHubMetadata::RepositoryFinder.new(new_site)
  end

  def self.repository
    @repository ||= GitHubMetadata::Repository.new(repository_finder.nwo).tap do |repo|
      Jekyll::GitHubMetadata.log :debug, "Generating for #{repo.nwo}"
    end
  end

  private
  attr_reader :repository_finder
    
  end
end
```

This enforces a safe top-level environment: when `site` is set to some other value, it resets everything and creates a new `repository_finder` instance with the new site.